src/TextInputWidget.cc

```diff
@@ -93,12 +97,50 @@ FilteredTextEdit::keyPressEvent(QKeyEvent *event)
                 }
                 break;
         }
+        case Qt::Key_V: {
+                if (event->modifiers() == Qt::ControlModifier) {
```

Comment:
Hardcoding this is incorrect. You cannot make assumptions about what keybind any given system uses to express "paste." Try using `QKeySequence::Copy` instead.

Reply:
On further review, given that Qt handles MIME clipboards in the first place, it's actually not obvious to me that handling pasting by hand like this is even necessary at all. If it truly is, some comments explaining why Qt's default behavior is inadequate would be nice.

Reply:
I'm unsure of what you mean. If you mean, that Qt has built in support out-of-the-box for pasting images into a QTextEdit, then I have not seen any document which suggests that. The fact that `QTextEdit::insertFromMimeData` and `QTextEdit::canInsertFromMimeData` exists suggests that it does not have built in support. However, this is just an educated guess.

Reply:
No need to guess. What happens if you disable this hardcoding but retain the overrides to those methods?

Reply:
Good call. I removed the explicit call to `insertFromMimeData()` (the entire case in the switch statement) and it still works.

Reply:
Fixed in `[41b887b](https://github.com/mujx/nheko/pull/138/commits/41b887b0554f36f527e2478d2bc205a0ae8fb232)`